src/vs/workbench/api/node/extHostWorkspace.ts

```diff
@@ -78,6 +78,17 @@ export class ExtHostWorkspace implements ExtHostWorkspaceShape {
 		}
 	}
 
+	addWorkspaceFolders(folders: { uri: URI, name?: string }[]): Thenable<void> {
+		return this._proxy.$addFolders(folders);
```

Comment:
Not yet decided if those changes should be reflected sync... We do that in some place, e.g the editor or Memento#update, where we assume optimistically that the change is good, then send to the main side which might cause another change... Needs some thinking.

Reply:
@jrieken well in some of the cases this action will result in the extension host restarting, so in that case it does not really matter. For the other case of an already opened workspace, I think even with the current promise chaining there is a chance that the `onDidChangeWorkspaceFolders` has not yet fired when the method returns.

What is your suggestion?